src/languages/php.md

```diff
@@ -140,6 +140,46 @@ This is the complete list of extensions that can be enabled:
 > see the up-to-date complete list of extensions after you SSH into
 > your environment. For PHP 7, use `ls /etc/php/*/mods-available`.
 
+## Custom PHP extensions
+
+It is possible to use an extension not listed here, but there are a few steps involved.
+
+1. Either download the extension .so file in your build hook or Git commit that to your Git repository.
+
+2. Provide a custom `php.ini` file in the application root (as a sibling of your `.platform.app.yaml` file) that loads the extension using an absolute path.  For example, if the extension is named `spiffy.so` and is in the root of your application, you would have a `php.ini` file that reads:
+
+```ini
+extension=/app/spiffy.so
+```
+
+3. Update your build flavor to none in your `.platform.app.yaml` file. You will need to provide a custom Composer command to use the extra extension. 
+
+```yaml
+build:
+    flavor: none
+```
+
+4. Add `PHP_INI_SCAN_DIR` environment variable and run `composer install` in your build hook.
+
+```yaml
+hooks:
+    build: |
+        export PHP_INI_SCAN_DIR="$(pwd):"
+        composer --no-ansi --no-interaction install --no-progress --prefer-dist --optimize-autoloader
+```
+
+If your extension is not in the application root then modify the directory accordingly.  (For instance, `export PHP_INI_SCAN_DIR="$(pwd)/my_extensions:"`)
+
+5. Then, in your `.platform.app.yaml` file, you also need the following in the web key so that PHP-FPM loads your .ini file first.
+
+```yaml
+web:
+    commands:
+        start: 'PHP_INI_SCAN_DIR="$(pwd):" /usr/sbin/php-fpm7.0'
```

Comment:
Hm.  Should we just ax these entirely, or move them to a separate add-on section for Ioncube?  Given that I really hate Ioncube and want to discourage its use I'm tempted to just ax entirely... :smile: Thoughts?